test/bats/test.bats

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bats
+
+load helpers
+
+BATS_TESTS_DIR=test/bats/tests
+
+@test "gatekeeper-controller-manager running" {
+  run kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod/gatekeeper-controller-manager-0
+  assert_success
+}
+
+@test "constrainttemplates crd is established" {
+  run kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh
+  assert_success
+}
+
+@test "apply config" {
+  run kubectl apply -f ${BATS_TESTS_DIR}/sync.yaml
+  assert_success
+}
+
+@test "required labels test" {
+  run kubectl apply -f ${BATS_TESTS_DIR}/templates/k8srequiredlabels_template.yaml
+  assert_success
+
+  while [[ -z $(kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/k8srequiredlabels.constraints.gatekeeper.sh) ]]; do echo "waiting for crd" && sleep 1; done
+
+  run kubectl apply -f ${BATS_TESTS_DIR}/constraints/all_ns_must_have_gatekeeper.yaml
+  assert_success
+
+  while [[ $(kubectl get k8srequiredlabels.constraints.gatekeeper.sh ns-must-have-gk -o json | jq '.status.byPod[].enforced') != "true" ]]; do echo "waiting for constraint" && sleep 1; done
```

Comment:
We should have a timeout on the waits that fail the test so we can move on to others and see if it's just one bad test, or all of them. A wait-with-timeout function could go in `helpers.bash`